pkg/oc/cli/cmd/rsync/copyrsync.go

```diff
@@ -111,6 +110,21 @@ func (r *rsyncStrategy) Validate() error {
 	return nil
 }
 
+// rsyncEscapeCsommand wraps each element of the command in double quotes
+// if it contains any of the following: single quote, double quote, backslash, or space
+// It also replaces every pre-existing double quote character in the element with a pair.
+// Example: " -> ""
+func rsyncEscapeCommand(command []string) []string {
+	for key := range command {
+		needsQuoted := strings.ContainsAny(command[key], `'" \`)
+		if needsQuoted {
+			command[key] = strings.Replace(command[key], `"`, `""`, -1)
```

Comment:
> I may need to brush up on my escaping, how does replacing a " with a "" help? wouldn't we want to replace a " with a \\" ?

I was confused too when I saw it.  It's an rsync thing.  From the rsync manpage,

> Note that doubling a single-quote inside a single-quoted string gives you a single-quote; likewise  for double-quotes (though you need to pay attention to which quotes your shell is parsing and which quotes rsync is parsing).

This is why in an earlier comment I suggested changing the name (to emphasize that this is rsync escaping, not shell escaping) and adding a comment to explain the rsync-style escaping.